cfn-resources/global-cluster-config/cmd/resource/resource.go

```diff
@@ -0,0 +1,365 @@
+
+	event = handler.ProgressEvent{
+		OperationStatus: handler.Success,
+		Message:         "Create Completed",
+		ResourceModel:   currentModel,
+	}
+	_, _ = logger.Debugf("Create() return event:%+v", event)
+	return event, nil
+}
+
+func Read(req handler.Request, prevModel *Model, currentModel *Model) (handler.ProgressEvent, error) {
+	setup() // logger setup
+	fmt.Printf("Read snapshot for Request() currentModel:%+v", currentModel)
+
+	event, client, err := ValidateRequest(currentModel)
+	if err != nil {
+		return event, err
+	}
+
+	// Check if  already exist
+	if !isExist(*client, currentModel) {
+		return handler.ProgressEvent{
+			OperationStatus:  handler.Failed,
+			Message:          "Resource Not Found",
+			HandlerErrorCode: cloudformation.HandlerErrorCodeNotFound}, nil
+	}
+	// method call to read configuration
+	config, _, err := ReadConfig(*client, currentModel)
+	if err != nil {
+		_, _ = logger.Debugf("error reading MongoDB Global Cluster Configuration (%s): %v", *currentModel.ClusterName, err)
+		return handler.ProgressEvent{
+			OperationStatus:  handler.Failed,
+			Message:          err.Error(),
+			HandlerErrorCode: cloudformation.HandlerErrorCodeNotFound}, nil
+	}
+	_, _ = logger.Debugf("Response Value:%+v", config)
+
+	return handler.ProgressEvent{
+		OperationStatus: handler.Success,
+		Message:         "Read Complete",
+		ResourceModel:   config,
+	}, nil
+}
+func ReadConfig(client mongodbatlas.Client, currentModel *Model) (*Model, handler.ProgressEvent, error) {
+	setup() // logger setup
+	projectID := *currentModel.ProjectId
+	ClusterName := *currentModel.ClusterName
+
+	// API call to read configuration
+	globalCluster, resp, err := client.GlobalClusters.Get(context.Background(), projectID, ClusterName)
+	if err != nil {
+		if resp != nil && resp.StatusCode == http.StatusNotFound {
+			return nil, handler.ProgressEvent{
+				OperationStatus:  handler.Failed,
+				Message:          err.Error(),
+				HandlerErrorCode: cloudformation.HandlerErrorCodeNotFound}, nil
+		}
+		return nil, handler.ProgressEvent{}, err
+	}
+	nameSpaces := globalCluster.ManagedNamespaces
+	zoneMappings := globalCluster.CustomZoneMapping
+
+	if len(nameSpaces) == 0 && len(zoneMappings) == 0 {
+		return nil, handler.ProgressEvent{
+			OperationStatus:  handler.Failed,
+			Message:          "resource Not Found",
+			HandlerErrorCode: cloudformation.HandlerErrorCodeNotFound}, nil
+	}
+	readModel := new(Model)
+	readModel.ProjectId = currentModel.ProjectId
+	readModel.ClusterName = currentModel.ClusterName
+	maps := flattenManagedNamespaces(globalCluster.ManagedNamespaces)
+	zones := customZoneToModelMappings(globalCluster.CustomZoneMapping)
+	readModel.CustomZoneMappings = zones
+	readModel.ManagedNamespaces = maps
+	readModel.ApiKeys = currentModel.ApiKeys
+	readModel.RemoveAllZoneMapping = currentModel.RemoveAllZoneMapping
+	_, _ = logger.Debugf("response Object:%+v", readModel)
+	return readModel, handler.ProgressEvent{}, nil
```

Comment:
you can move all of these inside a function `newReadModel(currentModel)` and then call it here r`eadModel := newReadModel(currentModel)`